doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,509 @@
+     described in :ref:`the present chapter <core-language>`, while
+     some others (like :n:`@term_if`) are language extensions that are
+     presented in :ref:`the next chapter <extensions>`.
+
+     .. insertprodn term qualid_annotated
+
+     .. prodn::
+        term ::= @term_product_or_fun
+        | @term_let
+        | @term_if
+        | @term_fix
+        | @term_cofix
+        | @term100
+        term100 ::= @term_cast
+        | @term10
+        term10 ::= @term_application
+        | @one_term
+        one_term ::= @ @qualid_annotated
+        | @term1
+        term1 ::= @term_projection
+        | @term_scope
+        | @term0
+        term0 ::= @qualid_annotated
+        | @sort
+        | @primitive_notations
+        | @term_evar
+        | @term_match
+        | @term_record
+        | @term_generalizing
+        | @term_ltac
+        | ( @term )
+        qualid_annotated ::= @qualid {? @univ_annot }
+
+     .. note::
+
+        Many :term:`commands <command>` and :term:`tactics <tactic>`
+        use :n:`@one_term` (in the syntax of their arguments) rather
+        than :n:`@term`.  The former need to be enclosed in
+        parentheses unless they're very simple, such as a single
+        identifier.  This avoids confusing a space-separated list of
+        terms or identifiers with a :n:`@term_application`.
+
+   type
+
+     To be valid and accepted by the |Coq| kernel, a term needs an
+     associated type.  We express this relationship by “:math:`x` *of
+     type* :math:`T`”, which we write as “:math:`x:T`”.  Informally,
+     “:math:`x:T`” can be thought as “:math:`x` *belongs to*
+     :math:`T`”.
+
+     A specificity of the |Coq| language is that types can depend on
+     terms (we say that the language is `dependently-typed
+     <https://en.wikipedia.org/wiki/Dependent_type>`_).  Because of
+     this, types and terms share a common syntax, all types are terms,
+     but not all terms are types:[#types]_
+
+     .. insertprodn type type
+
+     .. prodn::
+        type ::= @term
+
+     The kernel of |Coq| is a type checker that verifies that a term has
+     the expected type by applying a set of typing rules (see
+     :ref:`Typing-rules`).  If that's indeed the case, we say that the
+     term is :gdef:`well-typed`.
+
+     Intuitively, types may be viewed as sets containing terms.  We
+     say that a type is :gdef:`inhabited` if it contains at least one
+     term (i.e. if we can find a term which is associated to this
+     type).  We call such term a :gdef:`witness`.  Note that deciding
+     whether a type is inhabited is `undecidable
+     <https://en.wikipedia.org/wiki/Undecidable_problem>`_.
+
+     Formally, this gives rise to logical foundations which are not
+     the same as the well-known `"set theory"
+     <https://en.wikipedia.org/wiki/Set_theory>`_.  |Coq| is based on
+     the Calculus of Inductive Constructions which is a `"type theory"
+     <https://en.wikipedia.org/wiki/Type_theory>`_.
```

Comment:
I've removed the phrasing "this gives rise" and mixed our two wordings into something new. I've kept "Formally," because it answers to "Intuitively," in the paragraph above.